diff --git a/src/unix/pty.cc b/src/unix/pty.cc
index 8c5ab4e..178ef8f 100644
--- a/src/unix/pty.cc
+++ b/src/unix/pty.cc
@@ -67,6 +67,10 @@
 #include <sys/event.h>
 #include <sys/sysctl.h>
 #include <termios.h>
+#endif
+
+#if defined(NODE_PTY_IOS)
+extern char **environ;
 #endif
 
 /* NSIG - macro for highest signal + 1, should be defined */
@@ -304,7 +308,7 @@ Napi::Value PtyFork(const Napi::CallbackInfo& info) {
   winp.ws_xpixel = 0;
   winp.ws_ypixel = 0;
 
-#if !defined(__APPLE__)
+#if !defined(__APPLE__) || defined(NODE_PTY_IOS)
   // uid / gid
   int uid = info[6].As<Napi::Number>().Int32Value();
   int gid = info[7].As<Napi::Number>().Int32Value();
@@ -353,7 +357,7 @@ Napi::Value PtyFork(const Napi::CallbackInfo& info) {
 
   pid_t pid;
   int master;
-#if defined(__APPLE__)
+#if defined(__APPLE__) && !defined(NODE_PTY_IOS)
   int argc = argv_.Length();
   int argl = argc + 4;
   std::unique_ptr<char *, DelBuf> argv_unique_ptr(new char *[argl], DelBuf(argl));
